libs/crypto/factory.hpp

```diff
@@ -14,15 +14,20 @@
 #pragma once
 
 #include "log/logger.hpp"
-#include "crypto/openssl/EdDSASigner.hpp"
-#include "crypto/openssl/EdDSAVerifier.hpp"
+#include "crypto/crypto.hpp"
+#include "crypto/signer.hpp"
+#include "crypto/verifier.hpp"
 
 namespace concord::crypto {
+enum class Provider : uint16_t {
+#ifdef USE_OPENSSL
+  OpenSSL
+#endif
+};
 
-enum class Provider : uint16_t { OpenSSL };
-
+#ifdef USE_OPENSSL
```

Comment:
you don't need this ifdef, if USE_OPENSSL is not defined there will be no such provider and you won't compile.